languages/go/src/main/kotlin/io/vrap/codegen/languages/go/client/MethodRenderer.kt

```diff
@@ -305,42 +302,37 @@ class GoMethodRenderer constructor(
                     "nil" to statusCode
                 }
             }
-            .groupBy {
-                Key(it.first, (it.second.toInt() in (200..299)))
-            }
-            .mapValues {
-                entry ->
-                entry.value.map { it.second.toInt() }
-            }
             .map {
+                val isSuccess = it.second.toInt() in (200..399)
```

Comment:
```suggestion
                val isSuccess = it.second.toInt() in (200..299)
```

3xx codes indicate redirects. As they don't transport response data I would not treat them as success if the client returns them.